pyneuroml/plot/PlotMorphologyVispy.py

```diff
@@ -1176,7 +1177,25 @@ def create_instanced_meshes(meshdata, plot_type, current_view, min_width):
             parent=current_view.scene,
         )
         # TODO: add a shading filter for light?
+        shading_filter = ShadingFilter(
+            "smooth", shininess=1, ambient_light=(1, 1, 1, 0.6)
+        )
+        mesh.attach(shading_filter)
+
         assert mesh is not None
+
+        def attach_headlight(current_view):
+            light_dir = (0, 1, 0, 0)
+            shading_filter.light_dir = light_dir[:3]
+            initial_light_dir = current_view.camera.transform.imap(light_dir)
+
+            @current_view.scene.transform.changed.connect
+            def on_transform_change(event):
+                transform = current_view.camera.transform
+                shading_filter.light_dir = transform.map(initial_light_dir)[:3]
+                # print(shading_filter.light_dir)
+
+    attach_headlight(current_view)
```

Comment:
this needs to be indented so that it's called for each new mesh. At the moment, it's only being called once for all meshes, which means it's only being applied for the last mesh..

Reply:
That doesn't fix the issue though. I'd file an issue on the vispy repo and leave this for the moment. Digging into vispy internals is not the best use of our time.